lib/galaxy/managers/library_datasets.py

```diff
@@ -139,7 +139,10 @@ def _validate_and_parse_update_payload(self, payload):
                 val = validation.validate_and_sanitize_basestring(key, val)
                 validated_payload[key] = val
             if key in ('tags'):
-                val = validation.validate_and_sanitize_basestring_list(key, val)
+                if isinstance(val, list):
+                    val = validation.validate_and_sanitize_basestring_list(key, val)
+                else:
+                    val = validation.validate_and_sanitize_basestring(key, val)
```

Comment:
What about:

```python
                val = listify(val)
                val = validation.validate_and_sanitize_basestring_list(key, val)
```

I think that at line 108 it is assumed that this must be a list.

Reply:
Yes, but I don't want to change the incoming payload here. `tag1,tag2` would then become `['tag1,tag2']`, that seems worse.

Reply:
No, `listify` splits a string on commas.